client/client.js

```diff
@@ -3,23 +3,53 @@ var assert = require('assert');
 var boot = require('loopback-boot');
 var debug = require('debug')('strong-mesh-models:client');
 var loopback = require('loopback');
-var request = require('request');
+var serverService = require('./models/server-service');
+var serviceInstance = require('./models/service-instance');
 var url = require('url');
+var urlDefaults = require('strong-url-defaults');
 var util = require('util');
+var path = require('path');
+
+/**
+ * Mesh API Client
+ *
+ * @param {string} apiUrl URL for Mesh server. Formats supported include:
+ *   * http+unix:///full/path/to/socket
+ *   * http://host:port
+ *      - Uses 127.0.0.1 if host is ommitted
+ *      - Uses 8701 if port is ommitted
+ * @param {object} [options] Options object
+ * @param {string} [options.overridePath] Path to where API is mounted on remote
+ * server. This is normally `/api` however can be overridden if proxies or
+ * mounted at a different location.
+ * @constructor
+ */
+function Client(apiUrl, options) {
+  options = options || {};
+
+  // Useful in cases where the API requests are being proxies. This is used in
+  // Arc for the internal PM.
+  var apiBasePath = options.overridePath || '/api';
 
-function Client(apiUrl) {
   // Normalize the URI
   var endpoint = url.parse(apiUrl);
-  debug('normalize endpoint %j', endpoint);
-
-  if (endpoint.protocol === 'http+unix:') {
+  if (!endpoint.protocol) {
+    this.apiUrl = util.format(
+      'http://unix:%s:%s',
+      path.resolve(apiUrl), apiBasePath
+    );
+  } else if (endpoint.protocol === 'http+unix:') {
```

Comment:
The unix socket scheme used by request is not easy to parse and validate. Using a custom scheme of http+unix:// makes that a lot simpler. http+unix:// is also similar to the http+ssh:// syntax that us used in CLI. If we switch to accepting the request style schema then I will just need to move this code into the CLI which seems unnecessary.